bundles/org.eclipse.osgi/container/src/org/eclipse/osgi/internal/debug/EclipseDebugTrace.java

```diff
@@ -613,15 +613,8 @@ private void writeMessage(final Writer traceWriter, final FrameworkDebugTraceEnt
 	 * returned if the input string is <code>null</code>.
 	 */
 	private static String encodeText(final String inputString) {
-		if (inputString == null || inputString.indexOf(TRACE_ELEMENT_DELIMITER) < 0)
-			return inputString;
-		final StringBuilder tempBuffer = new StringBuilder(inputString);
-		int currentIndex = tempBuffer.indexOf(TRACE_ELEMENT_DELIMITER);
-		while (currentIndex >= 0) {
-			tempBuffer.replace(currentIndex, currentIndex + TRACE_ELEMENT_DELIMITER.length(), TRACE_ELEMENT_DELIMITER_ENCODED);
-			currentIndex = tempBuffer.indexOf(TRACE_ELEMENT_DELIMITER);
-		}
-		return tempBuffer.toString();
+		return inputString == null ? null
+				: inputString.replace(TRACE_ELEMENT_DELIMITER, TRACE_ELEMENT_DELIMITER_ENCODED);
```

Comment:
Looks like an appropriate reduction of boiler plate code!
I wonder if that code be a hotspot in any situation. Given that those Strings are in fact single characters it would be even more efficient to use the char version String.replace(char oldChar, char newChar).

Reply:
The encoded string has multiple characters, so that would not work.

Besides that I share your assessment and agree that it is a good improvement, mainly because of code simplification and if it improves performance in certain situations even better. :)